docs/Performance/async-await.md

```diff
@@ -0,0 +1,117 @@
+* [Spread Syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)
+* [Arrow function expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)
+* [Shorthand property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#new_notations_in_ecmascript_2015)
+
+```js
+/** 
+ * Returns the result of the OData call
+ * @param {object} payload - The payload which equals to the entity
+ * @param {object} parameters - The parameters like Filters added to the OData call
+ * @returns {Promise} Promise object represents result of the OData call
+ */
+function readMyEntity({ payload, parameters = {} }) {
+  const model = this.getView().getModel()
+  return model.metadataLoaded()
+    .then(() => {
+      return new Promise((resolve, reject) => {
+        const key = model.createKey('/MyEntitySet', payload)
+        // prevent success / error callbacks to be overwritten
+        const params = {
+          ...parameters,
+          success: (data, response) => {
+            // prevent accidently change of response data for subsequent calls
+            resolve({ data: { ...data }, response });
+          },
+          error: error => {
+            // additional error handling when needed
+            reject(error)
+          },
+        }
+        model.read(key, params)
+        // update could look like this:
+        // model.update(key, payload, params)
+        // create could look like this:
+        // model.create('/MyEntitySet', payload, params)
+        // delete could look like this:
+        // model.remove(key, params)
+      })
+    })
+}
+```
+
+### async / await syntax sugar
+
+```js
+async function init() {
+  try {
+    const { data, response } = await readMyEntity({ id: 12345 })
+    console.log(data, response)
+  } catch (error) {
+    console.error(error)
+  }
+}
+```
+
+### Run multiple Promises in parallel
+
+#### Promise.all
+
+Breaks with the first Promise that is rejected.
+
+<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all>
+
+```js
+async function init() {
+  try {
+    const [result1, result2] = await Promise.all([
+      readMyEntity({ id: 12345 }),
+      readMyEntity({ id: 67890 }),
+    ])
+    console.log(result1, result2)
+  } catch (error) {
+    console.error(error)
+  }
+}
+```
+
+#### Promise.allSettled
+
+Waits until all Promises are fullfilled and tells the status as well as the result.
+
+<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled>
+
+```js
+async function init() {
+  try {
+    const [result1, result2] = await Promise.allSettled([
+      readMyEntity({ id: 12345 }),
+      readMyEntity({ id: 67890 }),
+    ])
+    console.log(result1, result2)
+  } catch (error) {
+    console.error(error)
+  }
+}
+```
```

Comment:
while you're spoiling the reader with all'em async goodness, why not add some examples mention on the not-so-well-known 
- `Promise.any()` &rarr; client-side companion to a Multi-Origin Composition in SAP GW https://help.sap.com/doc/05d53b2d3bbb43d2ab5efa23829b2777/1610%20002/en-US/frameset.htm?bcf72651c294256ee10000000a445394.html 
  &rarr; "first service that answers (aka fulfills the promise) wins"
- `Promise.race()` &rarr; "first rejected or resolved promise wins"

of course those just came into my mind and don't have to be included in here :)

Reply:
I don't think this would bring value.
Both methods are coming from Node.js and might have more value in the backend as in the frontend.

I don't see the multi origin use frontend use case - this should happen transparently in the gw or middleware.
To do this in the frontend feels extremely fragile to me.